database.go

```diff
@@ -0,0 +1,26 @@
+package eremetic
+
+import "encoding/json"
+
+// TaskDB defines the functions needed by the database abstraction layer
+type TaskDB interface {
+	Clean() error
+	Close()
+	PutTask(task *Task) error
+	ReadTask(id string) (Task, error)
+	ReadUnmaskedTask(id string) (Task, error)
+	ListNonTerminalTasks() ([]*Task, error)
+}
+
+const Masking = "*******"
+
+func ApplyMask(task *Task) {
```

Comment:
fair enough. I can see the value in having patchsets only moving code
around and not changing stuff at the same time.

On Wed, 7 Sep 2016, 18:54 Marcus Olsson, <notifications@github.com> wrote:

> In database.go
> <https://github.com/klarna/eremetic/pull/129#discussion_r77862170>:
>
> > +
> > +import "encoding/json"
> > +
> > +// TaskDB defines the functions needed by the database abstraction layer
> > +type TaskDB interface {
> > +	Clean() error
> > +	Close()
> > +	PutTask(task *Task) error
> > +	ReadTask(id string) (Task, error)
> > +	ReadUnmaskedTask(id string) (Task, error)
> > +	ListNonTerminalTasks() ([]*Task, error)
> > +}
> > +
> > +const Masking = "*******"
> > +
> > +func ApplyMask(task *Task) {
>
> I agree, this PR (and the previous one) are really organizational changes
> and I've gone with the path of the least resistance here. I believe it will
> enable further refactoring work like the one you mentioned.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/klarna/eremetic/pull/129/files/66ce017b9d78aaf4e3947c587c11cf6b272b5111#r77862170>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHohjqsbs-kYdAUiJGMfPOoPLaIG2Kzks5qnuwmgaJpZM4J21At>
> .
>